shared/apply-patches.sh

```diff
@@ -60,13 +60,18 @@ if [ -d "$PATCHES_DIRECTORY" ]; then
     if [ ! -f "$PATCHES_DIRECTORY/$file" ]; then
       echo "Ignoring $PATCHES_DIRECTORY/$file, not a file"
     else
-      echo "Applying $PATCHES_DIRECTORY/$file to $ARGV_DESTINATION_DIRECTORY"
-      patch \
-        --force \
-        --directory="$ARGV_DESTINATION_DIRECTORY" \
-        --ignore-whitespace \
-        --strip=1 \
-        --input="$PATCHES_DIRECTORY/$file"
+      # Check if patch has already been applied
+      if [[ $(patch -sNp1 --dry-run < "$PATCHES_DIRECTORY/$file" &2>/dev/null) ]]; then
+        echo "Ignoring $PATCHES_DIRECTORY/$file, patch already applied"
+      else
+        echo "Applying $PATCHES_DIRECTORY/$file to $ARGV_DESTINATION_DIRECTORY"
+        patch \
+          --force \
+          --directory="$ARGV_DESTINATION_DIRECTORY" \
+          --ignore-whitespace \
+          --strip=1 \
+          --input="$PATCHES_DIRECTORY/$file"
```

Comment:
You don't need an if / else, just run `patch ... || echo "patch already applied"`

Reply:
@zvin The first command is a dry run, which means changes are not applied, it's just to check so we still need the if/else to actually apply the changes if the check fails

Reply:
That's why it is not needed, if `patch --dry-run fails`, `patch` would have failed too, we're just running it twice.